src/codeflare_sdk/cluster/cluster.py

```diff
@@ -472,7 +549,7 @@ def get_current_namespace():  # pragma: no cover
             return None
 
 
-def get_cluster(cluster_name: str, namespace: str = "default"):
+def get_cluster(cluster_name: str, namespace: str = "default", mcad=True):
```

Comment:
I think this needs to check for an appwrapper with the same cluster name to decide whether or not the param mcad=True. Otherwise, you could set up a raycluster only cluster earlier, then re-connect here without specifying mcad=False and generate an appwrapper yaml that was never used.

Reply:
The issue here is that the AppWrapper CRD may not (and most likely will not) even exist for users who are doing KubeRay direct. Can you test what currently happens if you specify the opposite version (mcad true when you did false, mcad false when you did true), and does it have any unexpected behavior?

Reply:
I think this is what I mean. A user who is using KubeRay direct will not have an associated appwrapper. In the case where there is no appwrapper, set `mcad=false`. I tested the case where I followed the KubeRay direct route, and then re-connected with `get_cluster(mcad=true)` and it overwrote my yaml file with an Appwrapper when it was deployed as just a RayCluster.yaml

Reply:
So the issue is that the user may not have the codeflare operator or MCAD installed at all, so in that case it won't be about a specific AW not existing, its the notion of an AW (the CRD itself with the associated API) won't exist. I guess we should just check for that or try to handle that case as well for auto-selection?